### ISSUE TITLE
ImportError: DLL load failed: The specified module could not be found.

### ISSUE THREAD

**Issue by rrajkumar77** [tensorflow/tensorflow#43033]
Name: tensorflow
Version: 2.3.0
Summary: TensorFlow is an open source machine learning framework for everyone.
Home-page: https://www.tensorflow.org/
Author: Google Inc.
Author-email: packages@tensorflow.org
License: Apache 2.0
Location: c:\users\user\anaconda3\lib\site-packages
Requires: numpy, wrapt, gast, protobuf, google-pasta, keras-preprocessing, tensorflow-estimator, termcolor, scipy, opt-einsum, tensorboard, absl-py, astunparse, wheel, grpcio, six, h5py
Required-by: 

---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
~\anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py in <module>
     63   try:
---> 64     from tensorflow.python._pywrap_tensorflow_internal import *
     65   # This try catch logic is because there is no bazel equivalent for py_extension.

ImportError: DLL load failed: The specified module could not be found.

During handling of the above exception, another exception occurred:

ImportError                               Traceback (most recent call last)
<ipython-input-3-ac9bda315bf2> in <module>
      1 # Importing the required libraries
----> 2 import tensorflow as tf
      3 
      4 
      5 # Import MNIST data

~\anaconda3\lib\site-packages\tensorflow\__init__.py in <module>
     39 import sys as _sys
     40 
---> 41 from tensorflow.python.tools import module_util as _module_util
     42 from tensorflow.python.util.lazy_loader import LazyLoader as _LazyLoader
     43 

~\anaconda3\lib\site-packages\tensorflow\python\__init__.py in <module>
     38 # pylint: disable=wildcard-import,g-bad-import-order,g-import-not-at-top
     39 
---> 40 from tensorflow.python.eager import context
     41 
     42 # pylint: enable=wildcard-import

~\anaconda3\lib\site-packages\tensorflow\python\eager\context.py in <module>
     33 from tensorflow.core.protobuf import config_pb2
     34 from tensorflow.core.protobuf import rewriter_config_pb2
---> 35 from tensorflow.python import pywrap_tfe
     36 from tensorflow.python import tf2
     37 from tensorflow.python.client import pywrap_tf_session

~\anaconda3\lib\site-packages\tensorflow\python\pywrap_tfe.py in <module>
     26 
     27 # pylint: disable=invalid-import-order,g-bad-import-order, wildcard-import, unused-import
---> 28 from tensorflow.python import pywrap_tensorflow
     29 from tensorflow.python._pywrap_tfe import *

~\anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py in <module>
     81 for some common reasons and solutions.  Include the entire stack trace
     82 above this error message when asking for help.""" % traceback.format_exc()
---> 83   raise ImportError(msg)
     84 
     85 # pylint: enable=wildcard-import,g-import-not-at-top,unused-import,line-too-long

ImportError: Traceback (most recent call last):
  File "C:\Users\user\anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 64, in <module>
    from tensorflow.python._pywrap_tensorflow_internal import *
ImportError: DLL load failed: The specified module could not be found.

Failed to load the native TensorFlow runtime.

See https://www.tensorflow.org/install/errors

for some common reasons and solutions.  Include the entire stack trace
above this error message when asking for help.

**Comment by Saduf2019** [comment]
@rrajkumar77 

Please installed the latest distribution of Visual C++ from [this link](https://support.microsoft.com/en-us/help/2977003/the-latest-supported-visual-c-downloads). and refer to below duplicate issues for the same error and let us know.
#41895 #42809 #43003 #42372 #42058

**Comment by rrajkumar77** [comment]
Even after installing Visual c++ i getting the same error.

---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
~\anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py in <module>
     63   try:
---> 64     from tensorflow.python._pywrap_tensorflow_internal import *
     65   # This try catch logic is because there is no bazel equivalent for py_extension.

ImportError: DLL load failed: The specified module could not be found.

During handling of the above exception, another exception occurred:

ImportError                               Traceback (most recent call last)
<ipython-input-7-2c56a628b4dd> in <module>
      1 # Importing the required libraries
----> 2 import tensorflow as tf
      3 
      4 # Import MNIST data
      5 mnist = tf.keras.datasets.mnist

~\anaconda3\lib\site-packages\tensorflow\__init__.py in <module>
     39 import sys as _sys
     40 
---> 41 from tensorflow.python.tools import module_util as _module_util
     42 from tensorflow.python.util.lazy_loader import LazyLoader as _LazyLoader
     43 

~\anaconda3\lib\site-packages\tensorflow\python\__init__.py in <module>
     38 # pylint: disable=wildcard-import,g-bad-import-order,g-import-not-at-top
     39 
---> 40 from tensorflow.python.eager import context
     41 
     42 # pylint: enable=wildcard-import

~\anaconda3\lib\site-packages\tensorflow\python\eager\context.py in <module>
     33 from tensorflow.core.protobuf import config_pb2
     34 from tensorflow.core.protobuf import rewriter_config_pb2
---> 35 from tensorflow.python import pywrap_tfe
     36 from tensorflow.python import tf2
     37 from tensorflow.python.client import pywrap_tf_session

~\anaconda3\lib\site-packages\tensorflow\python\pywrap_tfe.py in <module>
     26 
     27 # pylint: disable=invalid-import-order,g-bad-import-order, wildcard-import, unused-import
---> 28 from tensorflow.python import pywrap_tensorflow
     29 from tensorflow.python._pywrap_tfe import *

~\anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py in <module>
     81 for some common reasons and solutions.  Include the entire stack trace
     82 above this error message when asking for help.""" % traceback.format_exc()
---> 83   raise ImportError(msg)
     84 
     85 # pylint: enable=wildcard-import,g-import-not-at-top,unused-import,line-too-long

ImportError: Traceback (most recent call last):
  File "C:\Users\user\anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 64, in <module>
    from tensorflow.python._pywrap_tensorflow_internal import *
ImportError: DLL load failed: The specified module could not be found.


Failed to load the native TensorFlow runtime.

See https://www.tensorflow.org/install/errors

for some common reasons and solutions.  Include the entire stack trace
above this error message when asking for help.

**Comment by Saduf2019** [comment]
@rrajkumar77 
Please confirm you have verified with the issues shared, compatibility issues and missing dll files in case.

You might be facing this issue because of the following reasons

You are running 32-bit Python
Your CPU does not support AVX instructions. Please share the make and model of your CPU, so that we can verify this.
Please take a look at the [system requirements](https://www.tensorflow.org/install/pip#system-requirements) and check if you have the correct dependencies installed.
You need to install the MSVC 2019 redistributable
There is a library that is in a different location/not installed on your system that cannot be loaded.

Please verify with existing issues, as this is a reported issue.

**Comment by rrajkumar77** [comment]
OS Name	Microsoft Windows 10 Pro
Version	10.0.19041 Build 19041
Other OS Description 	Not Available
OS Manufacturer	Microsoft Corporation
System Name	DESKTOP-RAJ
System Manufacturer	Hewlett-Packard
System Model	HP EliteBook 840 G2
System Type	x64-based PC
System SKU	N2C61US#ABA
Processor	Intel(R) Core(TM) i7-5600U CPU @ 2.60GHz, 2601 Mhz, 2 Core(s), 4 Logical Processor(s)
BIOS Version/Date	Hewlett-Packard M71 Ver. 01.18, 9/26/2016
SMBIOS Version	2.7
Embedded Controller Version	150.91
BIOS Mode	Legacy
BaseBoard Manufacturer	Hewlett-Packard
BaseBoard Product	2216
BaseBoard Version	KBC Version 96.5B
Platform Role	Mobile
Secure Boot State	Unsupported
PCR7 Configuration	Binding Not Possible
Windows Directory	C:\WINDOWS
System Directory	C:\WINDOWS\system32
Boot Device	\Device\HarddiskVolume1
Locale	India
Hardware Abstraction Layer	Version = "10.0.19041.423"
User Name	DESKTOP-RAJ\user
Time Zone	India Standard Time
Installed Physical Memory (RAM)	8.00 GB
Total Physical Memory	7.88 GB
Available Physical Memory	943 MB
Total Virtual Memory	13.9 GB
Available Virtual Memory	2.67 GB
Page File Space	6.00 GB
Page File	C:\pagefile.sys
Kernel DMA Protection	Off
Virtualization-based security	Not enabled
Device Encryption Support	Reasons for failed automatic device encryption: PCR7 binding is not supported, Hardware Security Test Interface failed and device is not Modern Standby, Disabled by policy
Hyper-V - VM Monitor Mode Extensions	Yes
Hyper-V - Second Level Address Translation Extensions	Yes
Hyper-V - Virtualization Enabled in Firmware	No
Hyper-V - Data Execution Protection	Yes

MSVC 2019 redistributable installed

**Comment by Saduf2019** [comment]
@rrajkumar77
Can you please let me know if this is an anaconda installation? if yes, installation issues within the Anaconda environment are tracked in the [Anaconda repo](https://github.com/ContinuumIO/anaconda-issues/issues).

Could you please submit a new issue using [this link](https://github.com/ContinuumIO/anaconda-issues/issues) and fill in the template, so that the issue can be tracked there. Thanks!

**Comment by rrajkumar77** [comment]
Yes, I use anaconda 
#12028 raised for anaconda issue through the link provided

**Comment by Saduf2019** [comment]
Moving this to resolved status as issue is created in anaconda repo.
